blueprints/layered-profile.md

```diff
@@ -0,0 +1,274 @@
+
+#### Client Impact
+- Existing Go client methods will be updated for the `/deliveryservices` and `/servers` endpoints in order to write TO API tests for the exising endpoints.
+- All functions which get Parameters on Delivery Services or Servers must be changed to get the Parameters of all assigned Profiles in order.
+
+#### Data Model / Database Impact
+
+- A new Database Table `server_profiles` as described below, will be created:
+- A new Database Table `deliveryservice_profiles` as described below, will be created:
+```text
+            Table "traffic_ops.deliveryservice_profiles"
+     Column     |  Type                    | Collation | Nullable | Default
+----------------+--------------------------+-----------+----------+--------
+deliveryservice | bigint                   |           | not null |
+profile         | bigint                   |           | not null |
+order           | bigint                   |           | not null |
+last_updated    | timestamp with time zone |           | not null | now()
+Indexes:
+"pk_deliveryservice_profile" PRIMARY KEY(profile)
+Foreign-key constraints:
+"fk_deliveryservice" FOREIGN KEY (deliveryservice) REFERENCES deliveryservice(id)
+```
+
+```text
+            Table "traffic_ops.server_profiles"
+     Column    |  Type                    | Collation | Nullable | Default
+---------------+--------------------------+-----------+----------+--------
+ server        | bigint                   |           | not null |
+ profile       | bigint                   |           | not null |
+ order         | bigint                   |           | not null |
+ last_updated  | timestamp with time zone |           | not null | now() 
+Indexes:
+    "pk_server_profile" PRIMARY KEY(profile)
+Foreign-key constraints:
+    "fk_server" FOREIGN KEY (server) REFERENCES server(id)
+```
+
+### ORT Impact
+New set of profiles will be created and the profile-parameter relationship will change.
+
+### Traffic Monitor Impact
+No impact
+
+### Traffic Router Impact
+No impact
+
+### Traffic Stats Impact
+No impact
+
+### Traffic Vault Impact
+No impact
+
+### Documentation Impact
+All existing endpoints will need to be updated, along with the documentation explaining `layered_profile`.
+
+### Testing Impact
+Client/API integration tests should be updated to verify the `layered_profile` functionality on existing API `/deliveryservices`, `/servers` endpoints.
+
+### Performance Impact
+We do not anticipate any performance impact with layered_profile approach.
+
+### Security Impact
+We do not anticipate any impact on security.
+
+### Upgrade Impact
+- A Database Migration to:
+  - drop profile column in existing deliveryservice and server table
+  - insert existing server and DS profiles along with their order into the new tables(deliveryservice_profiles, server_profiles)
+- The new capability can just be added to the `seeds.sql` file.
```

Comment:
Adding two new tables (deliveryservices_profiles, server_profiles).

Reply:
But you said there's no new endpoint like e.g. `deliveryservices/{{ID}}/profiles`, so shouldn't the ability to modify DS/server Profiles be covered by `ds-write` and `server-write`, which today already include the ability to modify Profile assignments to those objects?

Reply:
umm.. seeds.sql includes changes to database.. how is there any connection to the API endpoints (which there are no new ones) and instead of the word capability (I can see the confusion), I will change it to tables.

Reply:
I think the new tables need to go into migrations, not `seeds.sql`